cmd/podman/compose.go

```diff
@@ -0,0 +1,264 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/common/pkg/config"
+	cmdMachine "github.com/containers/podman/v4/cmd/podman/machine"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/pkg/machine"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+var composeCommand = &cobra.Command{
+	Use:   "compose [options]",
+	Short: "Run compose workloads via an external provider such as docker-compose or podman-compose",
+	Long: `This command is a thin wrapper around an external compose provider such as docker-compose or podman-compose.  This means that podman compose is executing another tool that implements the compose functionality but sets up the environment in a way to let the compose provider communicate transparently with the local Podman socket.  The specified options as well the command and argument are passed directly to the compose provider.
+
+The default compose providers are docker-compose and podman-compose.  If installed, docker-compose takes precedence since it is the original implementation of the Compose specification and is widely used on the supported platforms (i.e., Linux, Mac OS, Windows).
+
+If you want to change the default behavior or have a custom installation path for your provider of choice, please change the compose_provider field in containers.conf(5).  You may also set PODMAN_COMPOSE_PROVIDER environment variable.`,
+	RunE:              composeMain,
+	ValidArgsFunction: completion.AutocompleteNone,
```

Comment:
This will not behave good with shell completion. I think we should come up with something more reasonable.
docker compose v2 is actually using cobra as well which means we could actually just pass through the hidden completion command logic, compare `docker-compose __complete ""`.
Of course that only works with docker-compose v2 but I assume that is what most would use here.

Alternatively we can try to parse --help of the compose commands but that seems way to fragile.

Reply:
Added completions now as you suggested.  Here and there, I see some garbage but it's mostly useful and an improvement.  Maybe we can iron out some of the warts.